### PR TITLE
Upgrade setCameraOrientation to setCameraPose

### PR DESCRIPTION
**Pull request by saihv**
setCameraPose now takes a pose argument instead of just orientation. Commanded pose is expected to contain position in NED, **relative to default camera location**. 

Status:

- [x] Base implementation of setCameraPose for Unreal
- [x] Testing in Unreal
- [x] Unity implementation
- [ ] Testing in Unity
- [ ] Documentation
- [ ] Replacements in examples/ros wrapper

### CODE REVIEW

File: AirLib/include/api/VehicleSimApiBase.hpp [hunk 1]
```diff
@@ -54,7 +54,7 @@ class VehicleSimApiBase : public msr::airlib::UpdatableObject {
     virtual const msr::airlib::Environment* getGroundTruthEnvironment() const = 0;
 
     virtual CameraInfo getCameraInfo(const std::string& camera_name) const = 0;
-    virtual void setCameraOrientation(const std::string& camera_name, const Quaternionr& orientation) = 0;
+    virtual void setCameraPose(const std::string& camera_name, const Pose& orientation) = 0;
```
Comment (rajat2004) [inline AirLib/include/api/VehicleSimApiBase.hpp:57]: `orientation` should be updated to `pose` here I think

**Comment by rajat2004** [comment]
Also need to change the function in Unity

File: AirLib/include/api/VehicleSimApiBase.hpp (hunk 1 above)
Comment (saihv) [reply AirLib/include/api/VehicleSimApiBase.hpp]: Done

**Comment by saihv** [comment]
@rajat2004 I added the corresponding Unity implementation: it would be great if you can give it a shot in Unity.

**Comment by rajat2004** [comment]
Yeah, will try to test it out today. It's a bit difficult to test Unity right now since it's currently broken in master and needs #2328 to get it running

**Comment by saihv** [comment]
@rajat2004 Hey, could you please give this a shot when you get a chance?

File: PythonClient/airsim/client.py [hunk 2]
```diff
@@ -155,17 +155,17 @@ def simGetCameraInfo(self, camera_name, vehicle_name = ''):
...
-    def simSetCameraOrientation(self, camera_name, orientation, vehicle_name = ''):
+    def simSetCameraPose(self, camera_name, pose, vehicle_name = ''):
         """
-        - Control the orientation of a selected camera
+        - Control the pose of a selected camera
 
         Args:
             camera_name (str): Name of the camera to be controlled
-            orientation (airsim.Quaternion()): Quaternion representing the desired orientation of the camera
+            orientation (airsim.Pose()): Pose representing the desired position and orientation of the camera
```
Comment (rajat2004) [inline PythonClient/airsim/client.py:164]: Should be `Pose` here

File: docs/image_apis.md [hunk 3]
```diff
@@ -166,9 +166,10 @@ To move around the environment using APIs you can use `simSetVehiclePose` API. T
 ## Camera APIs
 The `simGetCameraInfo` returns the pose (in world frame, NED coordinates, SI units) and FOV (in degrees) for the specified camera. Please see [example usage](https://github.com/Microsoft/AirSim/tree/master/PythonClient//computer_vision/cv_mode.py).
 
-The `simSetCameraOrientation` sets the orientation for the specified camera as quaternion in NED frame. The handy `airsim.to_quaternion()` function allows to convert pitch, roll, yaw to quaternion. For example, to set camera-0 to 15-degree pitch, you can use:
+The `simSetCameraPose` sets the pose for the specified camera while taking an input pose as a combination of relative position and a quaternion in NED frame. The handy `airsim.to_quaternion()` function allows to convert pitch, roll, yaw to quaternion. For example, to set camera-0 to 15-degree pitch while maintaining the same position, you can use:
 ```
-client.simSetCameraOrientation(0, airsim.to_quaternion(0.261799, 0, 0)); #radians
+camera_pose = airsim.Pose(airsim.Vector3(0, 0, 0), airsim.to_quaternion(0.261799, 0, 0))  #RPY in radians
```
Comment (rajat2004) [inline docs/image_apis.md:171]: `Vector3r`

**Comment by rajat2004** [comment]
@saihv Sorry for the delay. I had setup Unity in the weekend but forgot to test it out
I've opened https://github.com/saihv/AirSim/pull/2 which is a rebased version of this PR along with 1 extra commit which adds a missing `SetPose` implementation in Unity
The PR can't be merged right now, but probably after rebase should be mergable. Or would you prefer that I open the PR directly against the master branch of this repo?

Also, is there any example script which uses this right now? I tested that there wasn't any crash or something which happened on calling this but nothing more till now. There are also problems with the Unity API right now, so can't be completely sure. Will update about any further testing

**Comment by rajat2004** [comment]
@saihv I've opened #2710 since there were a lot of conflicts with your branch as well, it has your commits as well

**Comment by saihv** [comment]
Thanks @rajat2004  - haven't been able to get around to this again recently. Let's continue discussion in the updated PR

**Comment by saihv** [comment]
Closing in favor of https://github.com/microsoft/AirSim/pull/2710

**Pull request closed without merging**